src/js/jwt/modules/useChromeAuth.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+import { wipePostbackParamsThatAreNotForUs, getOfflineToken } from '../insights/offline';
+
+import flatten from 'lodash/flatten';
+
+import * as jwt from '../jwt';
+import cookie from 'js-cookie';
+import { options as defaultOptions, allowedUnauthedPaths } from '../constants';
+const TIMER_STR = '[JWT][jwt.js] Auth time';
+
+function getWindow() {
+  return window;
+}
+
+function bouncer() {
+  if (allowUnauthed()) {
+    return;
+  }
+
+  if (!jwt.isAuthenticated()) {
+    cookie.remove(defaultOptions.cookieName);
+    jwt.login();
+  }
+
+  console.timeEnd(TIMER_STR); // eslint-disable-line no-console
+}
+
+function getAllowedUnauthedPaths() {
+  return flatten(allowedUnauthedPaths.map((e) => [e, e + '/']));
+}
+
+export function allowUnauthed() {
+  if (getAllowedUnauthedPaths().includes(getWindow().location.pathname)) {
+    return true;
+  }
+
+  return false;
+}
+
+const useChromeAuth = () => {
```

Comment:
In order to truly re-use only one instance, I think we need to use React context. That would wrap the whole chrome and as a value provide the return object from this function. This implementation will run `const promise = jwt.init(options).then(bouncer);` and every other function call on each hook call. Which is something we probabylu don't want.

The `useChromeAuth` would then be just a wrapped useContext, consuming the auth context value.